controllers/users.js

```diff
@@ -1029,6 +1029,16 @@ const updateUsernames = async (req, res) => {
   }
 };
 
+const getUsersWithAbandonedTasks = async (req, res) => {
+  try {
+    const data = await userQuery.fetchUsersWithAbandonedTasks();
```

Comment:
why are we interacting with model directly? we have data service layer.

Reply:
I followed the convention currently used in the controller files, where database interaction is handled via models. The model function is then used in the controller and subsequently in routes.
Should I move this part to services instead? I noticed that other functions in the controllers are also following the same approach mentioned above

Reply:
we have both service and data-access layer.
It would be good if we separate the concerns.

Reply:
Are you suggesting to move the code in models/user to services/user?
The highlighted code is a controller which used the function in model to get the result.

Reply:
I am suggesting to follow controller, service, data-access layer pattern.
Data-access layer should use the model. Service should use data-access layer and so on.

Reply:
Okay got it. Will make this change in both PR's.

Reply:
Have added the changes @pankajjs